pcdsdevices/epics_motor.py

```diff
@@ -735,6 +737,27 @@ def flag_is_cleared(value=None, **kwargs):
             status_wait(st, timeout=timeout)
         return st
 
+    @property
+    def md(self):
+        if self._md is None:
+            raise AttributeError('Device does not have an attached md, '
+                                 'and was likely not initialized from happi')
+        return self._md
+
+    @md.setter
+    def md(self, new_md):
+        """ initialize attributes when md is set """
+        self._md = new_md
+        self._extra = self._md.extraneous
+        self._id = self._extra['_id']
+        self._stageidentity = self._extra['stageidentity']
```

Comment:
Correct me if I'm wrong, but `stageidentity` and `pvbase` seem to be fields you're assuming are in your `new_md` argument.  With such a general looking setter property, I'm worried others might try to set metadata information that does not fit this use case and cause errors.  

There's a few things you could do off the top of my head:
* Re-name these methods to make them more specific to the pmgr use-case  (or at least flesh out the docstring to describe this more spefically)
* add some logic to deal with missing keys
``` python
self._stageidentity = self._extra.get('stageidentity')  # returns None if key does not exist 
if self._stageidentity is None:
    # raise some error message?
    return
```

unfortunately `HappiItem` doesn't let you add info to it, so if you really wanted to preserve the container structure you'd have to make a new container.  I'm starting to think you might want to just pull the info you need from the happi container and store your pmgr metadata in a different field.

Reply:
I think the most likely `KeyError` here will be from someone adding an IMS motor to the non-questionnaire `happi`. I agree that this case needs to be handled.

Any thoughts on only allowing this to be set exactly once to avoid people trying to set over it?

Reply:
To provide a suggestion here as to how you might handle this, you could choose to only continue with this setter method if the `stageidentity` key exists.  My original suggestion from above still applies here.  

I think future modifications to `happi` are positioned to better solve the only-use-method-once problem.